File: 프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.py
def solution(n):
    x3 = []
    num_list = range(1,n+1)
    i = 0
    while len(x3) < len(num_list):
        i += 1
        if '3' not in str(i) and i%3 != 0:
            x3.append(i)
        
    return i

File: 프로그래머스/0/120871. 저주의 숫자 3/저주의 숫자 3.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int i = 1;
        while (i <= n) {
            answer++;
            String num = Integer.toString(answer);
            while (num.contains("3") || answer % 3 == 0) {
                answer++;
                num = Integer.toString(answer);
            }
            i++;
        }
        return answer;
    }
}